### PR TITLE
feat: add error reporting

### PR DESCRIPTION
**Pull request by dni**
- notify user when claiming failed
- i18n error messages
- add errors table to indexed db
- add ui into settings with download / copy errors

screenshot-1714894236

### CODE REVIEW

**Comment by kilrau** [comment]
Very nice!

- Maybe let's go for "copy" only for now, no download option, as download won't work on iOS without shenanigans (where we crucially need it)? Or have download on Desktop only.
- I'd rename to `Local Errors:` to make it crystal clear that nothing is being sent anywhere
- show all functions as icons as space is very limited and it feels cluttered with things written out.
- Padding is off on mobile
image

**Review by kilrau — CHANGES_REQUESTED** [review]
Very nice!

- Maybe let's go for "copy" only for now, no download option, as download won't work on iOS without shenanigans (where we crucially need it)? Or have download on Desktop only.
- I'd rename to `Local Errors:` to make it crystal clear that nothing is being sent anywhere
- show all functions as icons as space is very limited and it feels cluttered with things written out.
- Padding is off on mobile
image

File: src/i18n/i18n.ts [hunk 1]
```diff
@@ -178,9 +178,18 @@ const dict = {
         canary: "Canary",
         settings: "Settings",
         decimal_separator: "Decimal Separator",
-        denomination_tooltip: "Choose your preferred denomination: BTC or sats",
-        decimal_tooltip:
-            "Choose your preferred decimal separator: dot or comma",
+        denomination_tooltip: "Choose your preferred denomination: btc or sat",
```
Comment (michael1011) [inline src/i18n/i18n.ts:181]: Why lowercase BTC?

File: src/components/ErrorLog.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        const reports = await getReports();
+        clipboard(JSON.stringify(reports));
+    };
+
+    const download = async (evt: MouseEvent) => {
+        evt.stopPropagation();
+        downloadJson(
+            `boltz-errors-${Math.floor(Date.now() / 1000)}`,
+            await getReports(),
+        );
```
Comment (michael1011) [inline src/components/ErrorLog.tsx:37]: That is not gonna work on Safari on mobile

File: src/components/ErrorLog.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            &nbsp;
+            <span onClick={download} class="btn-small">
+                {t("download")}
+            </span>
+            &nbsp;
+            <span onClick={copy} class="btn-small">
+                {t("copy")}
+            </span>
+            &nbsp;
+            <span class="btn-small btn-danger">{count()}</span>
```
Comment (michael1011) [inline src/components/ErrorLog.tsx:54]: Why show the count explicitly?

File: src/components/RefundButton.tsx [hunk 4]
```diff
@@ -169,15 +168,18 @@ const RefundButton = ({
                             msg = t("locktime_not_satisfied");
                         }
                         log.error(msg);
-                        setNotification(msg);
+                        notify("error", msg);
+                        report(swap().id, msg, jsonError);
```
Comment (michael1011) [inline src/components/RefundButton.tsx:172]: I am afraid that catching errors that are being thrown here is not going to be good enough. Can we hook into the logging library and also record every log statement?

File: src/i18n/i18n.ts (hunk 1 above)
Comment (kilrau) [reply src/i18n/i18n.ts:181]: I already fixed this here: https://github.com/BoltzExchange/boltz-web-app/pull/549/commits/941ab52e9c12b51b22e984beff91671ea4a5a3c4

File: src/components/ErrorLog.tsx (hunk 3 above)
Comment (kilrau) [reply src/components/ErrorLog.tsx:54]: Agree, we can leave it out to make the UI cleaner. Doesn't add any value if my counter is 1 or 1243844.

**Comment by kilrau** [comment]
@michael1011 as just discussed with @dni we want a test to see if clean up after 1 week works

**Pull request merged**